### PR TITLE
💻💻 CLI password generation options cleanup

### PR DESCRIPTION
**Pull request by louib**
Summary of changes:
* extract function for creating password generator from options into
`Generate` command. This function is now reused in `Add` and `Edit`
commands.
* Updated manpage with missing password generation options.
* Updated manpage with missing longer forms of password generation options.
* Added unit tests for new password generation options in `Add` and
`Edit`.
* Handle case when `-g` and `-p` options are used at the same time.

This PR adds password generation functionalities while reducing
code duplication, but at the cost of 2 small breaking changes:
* the password generation option for `Add` and `Edit` for specifying
password length is now `-L` instead of `-l`, to not clash with the
`-l --lower` option.
* the `-u` shorthand for the `--upper` option had to be renamed to `-U`, to not
clash with the `-u --username` option.

@droidmonkey what about the  `CHANGELOG` file, should I update it with the breaking changes now, or do you wait before the releases to do that?

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix
- ✅ Refactor
- ✅ New feature
- ✅ Breaking change
- ✅ Documentation

## Testing strategy
existing + new unit tests.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: tests/TestCli.cpp [hunk 1]
```diff
@@ -224,9 +228,6 @@ void TestCli::testAdd()
                     "newuser2",
                     "--url",
                     "https://example.net/",
-                    "-g",
```
Comment (louib) [inline tests/TestCli.cpp]: @droidmonkey `-g` and `-p` options were used at the same time, and the password prompt was winning. This case is now explicitly handled by both `add` and `edit`.

**Comment by droidmonkey** [comment]
Please do update the CHANGELOG. You can use a "2.5.0-Beta1" title. I actually like the idea of gradually growing the log with major PR's so I don't miss any important bits later on.

**Comment by louib** [comment]
@droidmonkey I added the 2 breaking changes to the CHANGELOG as well as a line about the new options. Is it clear that :boom: means a breaking change? :laughing:

**Comment by drone540** [comment]
Why not keep the -u option for the --upper password generation, and make the --username option use -U, or vice versa (-U = upper, -u = username).  As it stands right now, it's missing a shorthand option if you wanted to do something like -ulnse to use all the groups.

**Comment by droidmonkey** [comment]
-U for upper makes a lot of sense

**Comment by louib** [comment]
@drone540 @droidmonkey good call, renamed to `-U` instead of removing the shorthand altogether.

**Comment by louib** [comment]
@phoerious @droidmonkey looks like the MacOS tests are hanging now :grimacing: :grimacing:

**Comment by droidmonkey** [comment]
Please ignore, the macOS server is currently down

**Pull request merged**